### PR TITLE
Merge Sequelize Models to Main

### PR DESCRIPTION
**Pull request by Xen0Xys**

### CODE REVIEW

File: src/database/models/comptability/accounts/AccountingAccount.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        },
+        name: {
+            type: DataTypes.STRING(255),
+            unique: true
+        },
+        amount: {
+            type: DataTypes.FLOAT,
+            defaultValue: 0,
+            validate: {
+                min: 0
```
Comment (Xen0Xys) [inline src/database/models/comptability/accounts/AccountingAccount.ts:18]: Accounting accounts can have a negative amount

File: src/database/models/comptability/products/MovementType.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import {DataTypes, Sequelize} from "sequelize";
+
+module.exports = (orm: Sequelize) => {
+    return orm.define('movement_type', {
```
Comment (Xen0Xys) [inline src/database/models/comptability/products/MovementType.ts:4]: No movement_type table in the model

**Review by Xen0Xys — COMMENTED** [review]
Dupplicate files:

- ProductMovement

File: src/database/models/comptability/transactions/personal/PersonalTransaction.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {DataTypes, Sequelize} from "sequelize";
+
+module.exports = (orm: Sequelize) => {
+    return orm.define('personal_transaction', {
+        code: {
+            type: DataTypes.STRING,
```
Comment (Xen0Xys) [inline src/database/models/comptability/transactions/personal/PersonalTransaction.ts:6]: Bad data type: `DataTypes.STRING(15)`

File: src/database/models/comptability/transactions/personal/PersonalTransaction.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        },
+        amount: {
+            type: DataTypes.FLOAT
+        },
+        flow: {
+            type: DataTypes.ENUM,
+            values: Object.values(TransactionFlow)
+        },
+        sale_code: {
+            type: DataTypes.STRING,
```
Comment (Xen0Xys) [inline src/database/models/comptability/transactions/personal/PersonalTransaction.ts:20]: Bad data type: `DataTypes.STRING(9)`

File: src/database/models/enums/PeriodicTransactionPeriod.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+enum PeriodicTransactionPeriod {
```
Comment (Xen0Xys) [inline src/database/models/enums/PeriodicTransactionPeriod.ts:1]: Maybe rename it `TransactionPeriod`

File: src/database/models/members/personal/PersonalAccount.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        id: {
+            type: DataTypes.INTEGER,
+            autoIncrement: true,
+            primaryKey: true
+        },
+        member_id: {
+            type: DataTypes.INTEGER
+        },
+        card_id: {
+            type: DataTypes.INTEGER,
```
Comment (Xen0Xys) [inline src/database/models/members/personal/PersonalAccount.ts:14]: Bad data type: `DataTypes.TEXT`

File: src/database/models/pictures/Picture.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true
+        },
+        type_id: {
+            type: DataTypes.INTEGER
+        },
+        name: {
+            type: DataTypes.STRING
```
Comment (Xen0Xys) [inline src/database/models/pictures/Picture.ts:14]: Bad data type: `DataTypes.STRING(255)`

File: src/database/models/pictures/Picture.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        },
+        path: {
+            type: DataTypes.STRING
```
Comment (Xen0Xys) [inline src/database/models/pictures/Picture.ts:17]: Bad data type: `DataTypes.TEXT`
Need to be unique

File: src/database/models/pictures/PictureType.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = (orm: Sequelize) => {
+    return orm.define('picture_type', {
+        id: {
+            type: DataTypes.INTEGER,
+            autoIncrement: true,
+            primaryKey: true
+        },
+        name: {
+            type: DataTypes.STRING
```
Comment (Xen0Xys) [inline src/database/models/pictures/PictureType.ts:11]: Bad data type: `DataTypes.STRING(255)`

File: src/database/models/products/Product.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+module.exports = (orm: Sequelize) => {
+    return orm.define("product", {
+        id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true
+        },
+        name: {
+            type: DataTypes.TEXT,
```
Comment (Xen0Xys) [inline src/database/models/products/Product.ts:11]: Bad data type: `DataTypes.STRING(255)`

File: src/database/models/products/Product.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+            type: DataTypes.FLOAT,
+            validate: {
+                min: 0
+            }
+        },
+        category_id: {
+            type: DataTypes.INTEGER
+        },
+        picture_id: {
+            type: DataTypes.INTEGER
```
Comment (Xen0Xys) [inline src/database/models/products/Product.ts:24]: Need to be nullable

File: src/database/models/products/ProductCategory.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+module.exports = (orm: Sequelize) => {
+    return orm.define("product_category", {
+        id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true
+        },
+        name: {
+            type: DataTypes.TEXT,
```
Comment (Xen0Xys) [inline src/database/models/products/ProductCategory.ts:11]: Bad data type: `DataTypes.STRING(255)`

File: src/database/models/sales/Sale.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import {DataTypes, Sequelize} from "sequelize";
+
+module.exports = (orm: Sequelize) => {
+    return orm.define('sale', {
+        code: {
+            type: DataTypes.STRING,
```
Comment (Xen0Xys) [inline src/database/models/sales/Sale.ts:6]: Bad data type: `DataTypes.STRING(9)`

File: src/database/models/sales/SaleType.ts [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = (orm: Sequelize) => {
+    return orm.define('sale_type', {
+        id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true
+        },
+        name: {
+            type: DataTypes.STRING
```
Comment (Xen0Xys) [inline src/database/models/sales/SaleType.ts:11]: Bad data type: `DataTypes.STRING(255)`
Need to be unique

**Review by loicmaes — APPROVED** [review]

**Pull request merged**